node/src/components/block_synchronizer/tests.rs

```diff
@@ -69,14 +77,91 @@ impl MockReactor {
         let ((_ancestor, reactor_event), _) = self.scheduler.pop().await;
         reactor_event
     }
+
+    async fn process_effects(&self, mut effects: Effects<Event>) -> Vec<MockReactorEvent> {
+        let mut events = Vec::new();
+        for effect in effects.drain(0..) {
+            tokio::spawn(async move { effect.await });
```

Comment:
Should we `await` the `JoinHandle` returned here so that we simulate a sequential execution of the effects in the queue? This would make the process predictable.

Reply:
We actually can't await here in this format of the `MockReactor` since we don't mock the other node components (Storage, ContractRuntime etc). Effects created through the effect builder create a responder that listens for the result. We don't want to wait for the result to come (because there is no code to respond). We just want a shortcut to assert which event is generated if the other components would have responded.